CMakeLists.txt

```diff
@@ -130,103 +210,174 @@ aux_source_directory(${CMAKE_CURRENT_SOURCE_DIR}/src SRCS)
-set(CPACK_RESOURCE_FILE_README ${CMAKE_CURRENT_SOURCE_DIR}/README)
-
-include(CPack)
-
 # Doxygen
+# Handled as extra target "doc"
+
 find_package(Doxygen)
 if(DOXYGEN_FOUND)
-  # fake autotools variables
-  set(DX_DOCDIR ${CMAKE_CURRENT_BINARY_DIR}/doc)
-  set(PACKAGE_VERSION ${CPACK_PACKAGE_VERSION})
-  set(srcdir ${CMAKE_CURRENT_SOURCE_DIR})
-  configure_file(resources/Doxyfile.in resources/Doxyfile @ONLY)
-  unset(DX_DOCDIR)
-  unset(PACKAGE_VERSION)
-  unset(srcdir)
-
-  add_custom_target(player_doc
-    ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/resources/Doxyfile
-    DEPENDS ${SRCS} ${MAIN_FILE}
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/resources
-    COMMENT "Generating API documentation with Doxygen" VERBATIM)
-  add_custom_target(doc)
-  add_dependencies(doc player_doc liblcf_doc)
+	# fake autotools variables
+	set(DX_DOCDIR ${CMAKE_CURRENT_BINARY_DIR}/doc)
+	set(srcdir ${CMAKE_CURRENT_SOURCE_DIR})
+	configure_file(resources/Doxyfile.in resources/Doxyfile @ONLY)
+	unset(DX_DOCDIR)
+	unset(PACKAGE_VERSION)
+	unset(srcdir)
+
+	add_custom_target(player_doc
+		${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/resources/Doxyfile
+		DEPENDS ${SRCS} ${MAIN_FILE}
+		WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/resources
+		COMMENT "Generating API documentation with Doxygen" VERBATIM)
+	add_custom_target(doc)
+	add_dependencies(doc player_doc)
+	if(PLAYER_BUILD_LIBLCF)
+		add_dependencies(doc liblcf_doc)
+	endif()
 endif()
 
 # test
-enable_testing()
-
-set(TEST_GAME_REPOSITORY_PATH "${CMAKE_CURRENT_SOURCE_DIR}/lib/TestGame")
-if(NOT EXISTS ${TEST_GAME_REPOSITORY_PATH})
-  execute_process(COMMAND ${GIT_EXECUTABLE} clone
-    "https://github.com/EasyRPG/TestGame.git"
-    "${TEST_GAME_REPOSITORY_PATH}")
-endif()
-set(TEST_GAME_PATH "${TEST_GAME_REPOSITORY_PATH}/TestGame-2000")
-
-set(TEST_ENVS "RPG_TEST_GAME_PATH=${TEST_GAME_PATH}")
-if(EXISTS $ENV{RPG_RTP_PATH})
-  get_filename_component(RPG_RTP_PATH "$ENV{RPG_RTP_PATH}" ABSOLUTE)
-  message(STATUS "using rtp path ${RPG_RTP_PATH}")
-  list(APPEND TEST_ENVS "RPG_RTP_PATH=${RPG_RTP_PATH}")
-endif()
-
-file(GLOB TEST_FILES ${CMAKE_CURRENT_SOURCE_DIR}/tests/*.cpp)
-foreach(i ${TEST_FILES})
-  get_filename_component(name "${i}" NAME_WE)
-  add_executable(test_${name} ${i})
-  target_link_libraries(test_${name} ${EASYRPG_PLAYER_LIBRARIES_ALL})
-  add_dependencies(test_${name} ${PROJECT_NAME}_Static)
-
-  add_test(
-    NAME test_${name} WORKING_DIRECTORY ${TEST_GAME_PATH}
-    COMMAND ${EXECUTABLE_OUTPUT_PATH}/test_${name})
-  set_tests_properties(test_${name}
-    PROPERTIES ENVIRONMENT "${TEST_ENVS}")
-endforeach()
+option(PLAYER_ENABLE_TESTS "Execute unit tests after compilation finishes" ON)
+
+if(PLAYER_ENABLE_TESTS)
+	enable_testing()
+
+	set(TEST_GAME_REPOSITORY_PATH "${CMAKE_CURRENT_SOURCE_DIR}/lib/TestGame")
+	if(NOT EXISTS ${TEST_GAME_REPOSITORY_PATH})
+		execute_process(COMMAND ${GIT_EXECUTABLE} clone
+			"https://github.com/EasyRPG/TestGame.git"
```

Comment:
btw. we could just use `--depth=1` here too as the history is not needed either.